js/views/inbox_view.js

```diff
@@ -258,12 +265,123 @@
       });
       this.listenTo(convoCollection, 'reset', removeAllConversations);
 
+      window.libsession
+        .getMessageQueue()
+        .events.addListener('success', this.handleMessageSentSuccess);
+
+      window.libsession
+        .getMessageQueue()
+        .events.addListener('fail', this.handleMessageSentFailure);
+
       Whisper.events.on('messageExpired', messageExpired);
       Whisper.events.on('userChanged', userChanged);
 
       // Finally, add it to the DOM
       this.$('.left-pane-placeholder').append(this.leftPaneView.el);
     },
+
+    async fetchHandleMessageSentData(m) {
+      // nobody is listening to this freshly fetched message .trigger calls
+      const tmpMsg = await window.Signal.Data.getMessageById(m.identifier, {
+        Message: Whisper.Message,
+      });
+
+      if (!tmpMsg) {
+        return null;
+      }
+
+      // find the corresponding conversation of this message
+      const conv = window.ConversationController.get(
+        tmpMsg.get('conversationId')
+      );
+
+      // then, find in this conversation the very same message
+      const msg = conv.messageCollection.models.find(
+        convMsg => convMsg.id === tmpMsg.id
+      );
+      return { conv, msg };
+    },
+
+    async handleMessageSentSuccess(m) {
+      const fetchedData = await this.fetchHandleMessageSentData(m);
+      if (!fetchedData) {
+        return;
+      }
+      const { msg, conv } = fetchedData;
+
+      const sentTo = msg.get('sent_to') || [];
+
+      const isOurDevice = window.libsession.Protocols.MultiDeviceProtocol.isOurDevice(
+        m.device
+      );
+
+      // Handle the sync logic here
+      if (!isOurDevice && !msg.get('synced') && !msg.get('sentSync')) {
+        const contentDecoded = textsecure.protobuf.Content.decode(
+          m.plainTextBuffer
+        );
+        const { dataMessage } = contentDecoded;
+        msg.sendSyncMessageOnly(dataMessage);
+
+        msg.set({ sentSync: true });
+      } else if (isOurDevice && msg.get('sentSync')) {
+        msg.set({ synced: true });
+      }
+      const primaryPubKey = await libsession.Protocols.MultiDeviceProtocol.getPrimaryDevice(
+        m.device
+      );
+      msg.set({
+        sent_to: _.union(sentTo, [primaryPubKey.key]),
+        sent: true,
+        expirationStartTimestamp: Date.now(),
+        // unidentifiedDeliveries: result.unidentifiedDeliveries,
+      });
+
+      await window.Signal.Data.saveMessage(msg.attributes, {
+        Message: Whisper.Message,
+      });
+      conv.updateLastMessage();
+
+      msg.trigger('sent', msg);
+    },
+
+    async handleMessageSentFailure(m, error) {
```

Comment:
Like above we can move majority of the logic here into `messages.js` too

Reply:
fixed